### ISSUE TITLE
Consensus crash-recovery

### ISSUE THREAD

**Issue by asonnino** [MystenLabs/narwhal#115]
### Description
To ensure crash-recovery of consensus, it is not enough to have the `CertificatesWaiter` to send to consensus every certificate for which their ancestors are in the storage. After crash-recovery, we may also need to send all parents certificates with rounds greater than `last_committed`. This means we need to boot the `CertificatesWaiter` with knowledge of the last committed round and modify its workings to potentially re-serve some past certificates as well.

Note that the Primary core also need to persist `last_voted` to not equivocate on certificates.

**Comment by huitseeker** [comment]
See also #17.

**Comment by asonnino** [comment]
fixed by https://github.com/MystenLabs/narwhal/pull/481

**Comment by gdanezis** [comment]
I do not think this is entirely done. As per the discussion with @akichidis - the interface between client <-> Bullshark has to be adapted to ensure upon recovery the stream of agreed messages is correct. See also (new) issue #730

**Comment by asonnino** [comment]
You are right! sorry for that
